assertk/src/commonMain/kotlin/assertk/assertions/list.kt

```diff
@@ -94,4 +94,64 @@ fun Assert<List<*>>.containsSubList(sublist: List<*>) = given { actual: List<*>
             show(actual)
         }"
     )
+}
+
+/**
+ * Asserts the list starts with the expected elements, in the same order.
+ *
+ * [1, 2, 3] startsWith [1, 2] pass
+ * [1, 2, 3] startsWith [2, 1] fails
+ * [1, 2, 3] startsWith [1, 2, 3] pass
+ * [] startsWith [1, 2] fails
+ * [1, 2] startsWith [] pass
+ * [] startsWith [] pass
+ *
+ * @see[endsWith]
+ */
+fun Assert<List<*>>.startsWith(vararg elements: Any?) = given { actual ->
+    val sublist = if (actual.size >= elements.size) {
+        actual.subList(0, elements.size)
+    } else {
+        actual
+    }
+
+    if (sublist.contentEquals(elements)) return
+
+    expected(
+        "to start with:${
+            show(elements)
+        }, but was:${
+            show(sublist)
+        }"
+    )
+}
+
+/**
+ * Asserts the list ends with the expected elements, in the same order.
+ *
+ * [1, 2, 3] endsWith [2, 3] pass
+ * [1, 2, 3] endsWith [3, 2] fails
+ * [1, 2, 3] endsWith [1, 2, 3] pass
+ * [] endsWith [1, 2] fails
+ * [1, 2] endsWith [] pass
+ * [] endsWith [] pass
+ *
+ * @see[startsWith]
+ */
+fun Assert<List<*>>.endsWith(vararg elements: Any?) = given { actual ->
+    val sublist = if (actual.size >= elements.size) {
+        actual.subList(actual.size - elements.size, actual.size)
+    } else {
+        actual
+    }
+
+    if (sublist.contentEquals(elements)) return
+
+    expected(
+        "to end with:${
```

Comment:
Hm, the other assertions will show the entire list for context which I think is helpful. I think wouldn't be a problem to do that with `startsWith` but might be hard to read with `endsWith`, maybe showing the sublist in addition if necessary?

```kotlin
if (isSublist) {
  expected("to end with:${show(elements)} but was ${show(actual)}\n ending not expected:${show(sublist)}")
} else {
  expected("to end with:${show(elements)} but was ${show(actual)}")
}
```
you can see similar assertions in iterable.kt.

Thoughts?

Reply:
Maybe `expected("to end with:${show(elements)} but was ${show(sublist)} in ${show(actual)}` for both cases? This is some duplication, but you have sublist at the beginning to verify what's wrong, and full list at the end for context

Reply:
Sounds good

Reply:
changed